### PR TITLE
Only format smarty aliases as money if specified

### PR DESCRIPTION
**Pull request by eileenmcnaughton**
Overview
----------------------------------------
Only format smarty aliases as money if specified

Before
----------------------------------------
When defining a token that should be assigned as a smarty variable in a smarty template it is always assigned money formatted, if it is money

After
----------------------------------------
When defining a token that should be assigned as a smarty variable in a smarty template formatting should be specified

Technical Details
----------------------------------------
Smarty aliases allow tokens to be assigned to the template as smarty variables. They are very new,
have only been implemented in these 2 places and are not accessible from outside of core, which
both have unit test cover.

When trying to extend it to cover another variable (totalTaxAmount) in another template I found
that a raw format is appropriate. I think it makes sense to explicitly specify.

Comments
----------------------------------------

### CODE REVIEW

**Comment by eileenmcnaughton** [comment]
test this please

File: Civi/Token/TokenCompatSubscriber.php [hunk 1]
```diff
@@ -70,9 +70,17 @@ public function onRender(TokenRenderEvent $e): void {
     if ($useSmarty) {
       $smartyVars = [];
       foreach ($e->context['smartyTokenAlias'] ?? [] as $smartyName => $tokenName) {
-        $smartyVars[$smartyName] = \CRM_Utils_Array::pathGet($e->row->tokens, explode('.', $tokenName), $e->context['locale'] ?? NULL);
+        $tokenParts = explode('|', $tokenName);
+        $modifier = $tokenParts[1] ?? '';
+        $smartyVars[$smartyName] = \CRM_Utils_Array::pathGet($e->row->tokens, explode('.', $tokenParts[0]), $e->context['locale'] ?? NULL);
         if ($smartyVars[$smartyName] instanceof \Brick\Money\Money) {
-          $smartyVars[$smartyName] = \Civi::format()->money($smartyVars[$smartyName]->getAmount(), $smartyVars[$smartyName]->getCurrency());
+          if ($modifier === 'crmMoney') {
```
Comment (totten) [inline Civi/Token/TokenCompatSubscriber.php:77]:
```suggestion
          // TODO: We should reuse the filters from TokenProcessor::filterTokenValue()
          if ($modifier === 'crmMoney') {
```

**Comment by eileenmcnaughton** [comment]
@seamuslee001 I'm confused - it says you closed this 'in #22560' - but I can't see any action you did on there that closed it - I've re-opened & re-based

**Comment by demeritcowboy** [comment]
It's because the description there uses a magic word followed the pr number, so github figures it means to autoclose it. Or maybe there is some plugin enabled that recognizes that phrase.

**Comment by eileenmcnaughton** [comment]
@demeritcowboy ah that makes sense

**Comment by demeritcowboy** [comment]
This seems to run ok.

Maybe out of scope since it was like this before the PR but while the email itself has money formatting, the amount that appears on the email activity copy on the contact is raw. I'm not sure if it's always been like that.

**Comment by eileenmcnaughton** [comment]
Thanks @demeritcowboy - I'm a bit surprised by that but hopefully I'll track it down....

**Pull request merged**